gc/base/ObjectScanner.hpp

```diff
@@ -55,15 +55,15 @@ class GC_ObjectScanner : public MM_BaseVirtual
 protected:
 	static const intptr_t _bitsPerScanMap = sizeof(uintptr_t) << 3;
 
-	omrobjectptr_t const _parentObjectPtr;	/**< Object being scanned */
 	uintptr_t _scanMap;						/**< Bit map of reference slots in object being scanned (32/64-bit window) */
+	fomrobject_t *_scanPtr;					/**< Pointer to base of object slots mapped by current _scanMap */
+	GC_SlotObject _slotObject;				/**< Create own SlotObject class to provide output */
```

Comment:
Can you document why the order of the fields changed?  If there was a reason it would be beneficial to document it in the comments.

Reply:
After _scanMap, these fields are the hottest fields in the object scanner, so I moved them closer to _scanMap to ensure that they would be more likely to be contained in same cache line. 

I agree that more detailed object scanner documentation is required. @rwy0717 is pushing that envelope as well, and will likely force implementation of https://github.com/eclipse/omr/issues/1638 at least. I intend to write a boxnote to explain how to implement object scanners for languages with alignment and reference encoding special needs. For now I will  include a synopsis of this in ObjectScanner.hpp comment. Beyond that I suggest we treat this (object scanner documentation) as a separate issue.